src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -138,49 +105,14 @@ jint JNI_OnLoad(JavaVM * jvm, void * reserved)
     ChipLogProgress(Controller, "Loading Java class references.");
 
     // Get various class references need by the API.
-    err = JniReferences::GetInstance().GetClassRef(env, "chip/devicecontroller/AndroidChipStack", sAndroidChipStackCls);
-    SuccessOrExit(err);
     err = JniReferences::GetInstance().GetClassRef(env, "chip/devicecontroller/ChipDeviceControllerException",
                                                    sChipDeviceControllerExceptionCls);
     SuccessOrExit(err);
     ChipLogProgress(Controller, "Java class references loaded.");
 
-    // Initialize the CHIP System Layer.
-    err = sSystemLayer.Init();
+    err = AndroidChipPlatformJNI_OnLoad(jvm, reserved);
```

Comment:
Because we can not make a so in platform/android, this will make 2 instances in CHIPController.so and CHIPPlatform.so, all values will not shared in them.